### ISSUE TITLE
[Rule proposal] Add limitation on a number of conditions in if block

### ISSUE THREAD

**Issue by alexey1966** [eslint/eslint#14909]
**Please describe what the rule should do:**

This rule will check number of conditions in every if statement. If number of conditions is more then specified amount (for example 2), trigger an error - “Please wrap your complex conditions into flags”. This will help code readability.

**What new ECMAScript feature does this rule relate to?**

No 

**What category of rule is this? (place an "X" next to just one item)**

Code readability and code maintainability.

**Provide 2-3 code examples that this rule will warn about:**

```js

if (x === 1 && y === 3 && z === 4) // trigger error, because number of conditions is more then specified (in this example - 2).

```

**Why should this rule be included in ESLint (instead of a plugin)?**

It’s related to another standard rules like max number of lines in file or max number of statements in function.

**Are you willing to submit a pull request to implement this rule?**

Yes, I am willing to submit a new pull request.

**Comment by aladdin-add** [comment]
Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept new rules if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify. You can always make a custom rule and publish a plugin.
